sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala

```diff
@@ -182,4 +182,12 @@ class HiveOrcSourceSuite extends OrcSuite with TestHiveSingleton {
       }
     }
   }
+
+  test("Enforce direct encoding column-wise selectively") {
+    Seq(true, false).foreach { convertMetastore =>
+      withSQLConf(HiveUtils.CONVERT_METASTORE_ORC.key -> s"$convertMetastore") {
+        testSelectiveDictionaryEncoding(isSelective = false)
```

Comment:
So even with `CONVERT_METASTORE_ORC` as true, we still can't use selective direct encoding?

Reply:
Yep. This is based on the current behavior which is a little related to your CTAS PR. Only read-path works as expected.

Reply:
When we change Spark behavior later, this test will be adapted according to it.